common/src/jni/main/cpp/org_conscrypt_NativeCrypto.cpp

```diff
@@ -3898,14 +3897,14 @@ static void NativeCrypto_X509_verify(JNIEnv* env, jclass, jlong x509Ref, jobject
     EVP_PKEY* pkey = fromContextObject<EVP_PKEY>(env, pkeyRef);
     JNI_TRACE("X509_verify(%p, %p)", x509, pkey);
 
-    if (x509 == nullptr) {
-        Errors::jniThrowNullPointerException(env, "x509 == null");
-        JNI_TRACE("X509_verify(%p, %p) => x509 == null", x509, pkey);
+    if (pkey == nullptr) {
```

Comment:
Not sure I totally follow why you've switched the order of these checks ... can you elaborate?